src/main/scala/shine/DPIA/showDPIA.scala

```diff
@@ -42,40 +43,40 @@ case class showDPIA(showTypes : ShowTypes = Off) {
   }
   def showPrim : Primitive[_] => String = {
     case Comment(s) => s"/* ${s} */"
-    case Seq(c1, c2) => s"{\n${tabulate(showPhrase(c1))};\n${tabulate(showPhrase(c2))}\n}"
-    case New(dt, f) => s"new (exp[${showDataType(dt)}] x acc[${showDataType(dt)}]) in ${showPhrase(f)}"
-    case Assign(dt, lhs, rhs) => s"${showTypedPhrase(DataKind, dt, lhs)} = ${showPhrase(rhs)}"
-    case IdxAcc(n, dt, idx, array) => s"${showTypedPhrase(DataKind, ArrayType(n, dt), array)}[${showPhrase(idx)}]"
-    case Idx(n, dt, idx, array) => s"${showTypedPhrase(DataKind, ArrayType(n, dt), array)}[${showPhrase(idx)}]"
-    case NatAsIndex(n, e) => s"${showTypedPhrase(NatKind, n, e)}"
-    case Drop(n, m, dt, e) => s"drop ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
-    case DropAcc(n, m, dt, e) => s"drop ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
-    case Take(n, m, dt, e) => s"take ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
-    case TakeAcc(n, m, dt, e) => s"take ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
-    case f : For => s"for ${showNat(f.n)}\n${tabulate(showPhrase(f.loopBody))}\n"
-    case f : ForNat => s"forNat ${showNat(f.n)}\n${tabulate(showPhrase(f.loopBody))}\n"
+    case Seq(c1, c2) => s"${showPhrase(c1)};\n${showPhrase(c2)}"
+    case New(dt, f) => s"(new (exp[${showDataType(dt)}] x acc[${showDataType(dt)}]) in \n${tab(showPhrase(f))}\n)\n"
+    case Assign(_, lhs, rhs) => s"${showTypedPhrase(lhs)} = ${showPhrase(rhs)}"
+    case IdxAcc(_, _, idx, array) => s"${showTypedPhrase(array)}[${showPhrase(idx)}]"
+    case Idx(_, _, idx, array) => s"${showTypedPhrase(array)}[${showPhrase(idx)}]"
+    case NatAsIndex(_, e) => showTypedPhrase(e)
+    case Drop(n, _, _, e) => s"(drop ${showNat(n)} ${showTypedPhrase(e)})"
+    case DropAcc(n, _, _, e) => s"(drop ${showNat(n)} ${showTypedPhrase(e)})"
+    case Take(n, _, _, e) => s"(take ${showNat(n)} ${showTypedPhrase(e)})"
+    case TakeAcc(n, _, _, e) => s"(take ${showNat(n)} ${showTypedPhrase(e)})"
+    case f : For => s"(for ${showNat(f.n)}\n${tab(showPhrase(f.loopBody))}\n)\n"
+    case f : ForNat => s"(forNat ${showNat(f.n)}\n${tab(showPhrase(f.loopBody))}\n)\n"
```

Comment:
I guess the parenthesis are not required here anymore since they are in `showTypedPhrase`?

Reply:
We can always improve this printer iteratively in the future.

Reply:
good point